pkg/analysis/commentstart/analyzer.go

```diff
@@ -51,6 +51,8 @@ func checkField(pass *analysis.Pass, field *ast.Field, tagInfo extractjsontags.F
 		fieldName = field.Names[0].Name
 	} else if ident, ok := field.Type.(*ast.Ident); ok {
 		fieldName = ident.Name
+	} else if selector, ok := field.Type.(*ast.SelectorExpr); ok {
+		fieldName = selector.Sel.Name
```

Comment:
This feels like it's going to be useful to many of the analyzers, probably want a helper at some point

Reply:
It seems this still can't handle pointer types like `*Type`. Let me check it.
If the logic  gets complex, it would be good to prepare a helper now.

Reply:
types.ExprString may work for this! Very useful function.
https://pkg.go.dev/go/types#ExprString